sklearn/decomposition/_base.py

```diff
@@ -57,26 +62,36 @@ def get_precision(self):
         precision : array, shape=(n_features, n_features)
             Estimated precision of data.
         """
+        xp, _ = get_namespace(self.components_)
+
         n_features = self.components_.shape[1]
 
         # handle corner cases first
         if self.n_components_ == 0:
-            return np.eye(n_features) / self.noise_variance_
+            return xp.eye(n_features) / self.noise_variance_
 
-        if np.isclose(self.noise_variance_, 0.0, atol=0.0):
-            return linalg.inv(self.get_covariance())
+        if self.noise_variance_ == 0.0:
+            return xp.linalg.inv(self.get_covariance())
 
         # Get precision using matrix inversion lemma
         components_ = self.components_
         exp_var = self.explained_variance_
         if self.whiten:
-            components_ = components_ * np.sqrt(exp_var[:, np.newaxis])
-        exp_var_diff = np.maximum(exp_var - self.noise_variance_, 0.0)
-        precision = np.dot(components_, components_.T) / self.noise_variance_
-        precision.flat[:: len(precision) + 1] += 1.0 / exp_var_diff
-        precision = np.dot(components_.T, np.dot(linalg.inv(precision), components_))
+            components_ = components_ * xp.sqrt(exp_var[:, np.newaxis])
+        exp_var_diff = exp_var - self.noise_variance_
+        exp_var_diff = xp.where(
+            exp_var > self.noise_variance_, exp_var_diff, xp.zeros_like(exp_var)
+        )
+        precision = components_ @ components_.T / self.noise_variance_
+        # TODO use views instead?
+        xp.reshape(precision, shape=(-1,), copy=False)[:: len(precision) + 1] += (
+            1.0 / exp_var_diff
+        )
```

Comment:
This is currently braking the tests. I plan to implement the proposal of
https://github.com/scikit-learn/scikit-learn/pull/26315/files#r1186082492 in the wrapper to fix it.